game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProResourceTracker.java

```diff
@@ -18,7 +18,7 @@ public ProResourceTracker(final GamePlayer player) {
 
   public ProResourceTracker(final int pus, final GameState data) {
     resources = new IntegerMap<>();
-    resources.add(data.getResourceList().getResource(Constants.PUS), pus);
+    resources.add(data.getResourceList().getResource(Constants.PUS).orElse(null), pus);
```

Comment:
@BBlaza Same, either the method assumes the value must be available, then new method `getResourceOrThrow?` or a default value should be used, e.g., `orElse(0)`.